components/builder-api-client/src/builder.rs

```diff
@@ -173,57 +170,60 @@ impl BuilderAPIClient {
         Self::new(endpoint, product, version, fs_root_path).map(|c| Box::new(c) as _)
     }
 
-    fn maybe_add_authz<'a>(&'a self,
-                           rb: RequestBuilder<'a>,
-                           token: Option<&str>)
-                           -> RequestBuilder<'_> {
+    fn maybe_add_authz(&self, rb: RequestBuilder, token: Option<&str>) -> RequestBuilder {
         if token.is_some() {
-            rb.header(Authorization(Bearer { token: token.unwrap().to_string(), }))
+            rb.bearer_auth(token.unwrap().to_string())
         } else {
             rb
         }
     }
 
-    fn add_authz<'a>(&'a self, rb: RequestBuilder<'a>, token: &str) -> RequestBuilder<'_> {
-        rb.header(Authorization(Bearer { token: token.to_string(), }))
+    fn add_authz(&self, rb: RequestBuilder, token: &str) -> RequestBuilder {
+        rb.bearer_auth(token.to_string())
     }
 
-    fn download<'a>(&'a self,
-                    rb: RequestBuilder<'a>,
-                    dst_path: &Path,
-                    token: Option<&str>,
-                    progress: Option<<BuilderAPIClient as BuilderAPIProvider>::Progress>)
-                    -> Result<PathBuf> {
-        let mut res = self.maybe_add_authz(rb, token).send()?;
-
-        debug!("Response: {:?}", res);
-
-        if res.status != hyper::status::StatusCode::Ok {
-            return Err(err_from_response(res));
+    fn download(&self,
+                rb: RequestBuilder,
+                dst_path: &Path,
+                token: Option<&str>,
+                progress: Option<<BuilderAPIClient as BuilderAPIProvider>::Progress>)
+                -> Result<PathBuf> {
+        debug!("Downloading file to path: {:?}", dst_path);
+        let mut resp = self.maybe_add_authz(rb, token).send()?;
+        debug!("Status: {:?}", resp.status());
+
+        if resp.status() != StatusCode::OK {
+            return Err(err_from_response(resp));
         }
 
         fs::create_dir_all(&dst_path)?;
 
-        let file_name = res.headers
-                           .get::<XFileName>()
-                           .expect("XFileName missing from response")
-                           .to_string();
+        let file_name = resp.headers()
+                            .get(X_FILENAME)
+                            .expect("XFileName missing from response")
+                            .to_str()
+                            .expect("Invalid X-Filename");
```

Comment:
I notice this pattern is repeated a lot, where in order to retrieve the value of some header, we have to call `headers().get(NAME_OF_HEADER).expect("something's missing").to_str().expect("Invalid something or other")`.

It'd be nice if there was a way to abstract that out into something that could be reused every time we need to get a header value out, ideally without the need to call `expect()` twice, but at the very least just to avoid all the repetition.

Reply:
Good point, will look at that